### DISCUSSION TITLE
Go back to front of selection after "Grabbing" for beacon?

### DISCUSSION THREAD

**Discussion by trev-dev** [meow-edit/meow#288] [Ideas]
I feel like it's not so intuitive to have the point stay at the _end_ of the selection after grabbing text and being thrown into beacon mode. I immediately feel the need to back-track or work in reverse if I'm preforming a macro. I feel like it ads to the clumsiness of beacon-mode.

Thoughts?

**Comment by eshrh** [comment]
We must have the same mind, I completely agree. My first command after entering beacon mode is in practice nearly always `C-a`, and i've always felt it was clumsy but i've been afraid that nobody else thought the same way and i was just using it wrong.

I have no idea how to change this in a reasonable way though. Jump the cursor to the min of point and mark on beacon entry? Another option is to kind of find the left boundary of the beacon, and then jump to the beginning of the secondary-selection if it's on the same line, or the beginning of the line if it's on multiple. I think that's even more counter intuitive. Besides, the cases in which you'd like to be at the front of the beacon are (at least for me) mostly after selection via meow-line and some number of expansions. Inconsistent behavior would be unacceptable.

**Reply by trev-dev** [reply]
Yeah consistency is definitely key for strong habits and muscle memory.

**Comment by DogLooksGood** [comment]
I don't think jumping cursor while grabbing is a good idea. You can do `meow-reverse` before `meow-grab`.

**Reply by trev-dev** [reply]
I didn't realize that worked! Explicit is better than implicit :)

**Reply by DogLooksGood** [reply]
Yep, `meow-reverse` is quite useful. You can imagine in this case: you have a command at `x`, and a reversed command at `Alt-x`. And `meow-reverse` is just that `Alt` key, but there's no need to hold down.

**Reply by eshrh** [reply]
Neat!!

**Reply by trev-dev** [reply]
I just made a selection from the "bottom-up" (in reverse) and it `meow-grab`. Sure enough, the cursor stays at the top of the selection that way, too.

Having direction be a more explicit paradigm may take some getting use to, but if I recall correctly, choosing direction is part of Meow's core philosophy.

**Reply by DogLooksGood** [reply]
Yeah, for character and word movements, we have individual commands for each direction, because they are frequently used. But for other movements, we only have one command for each. Because good keys are few. In my opinion, pressing a key with Shift, is as hard as pressing two keys.

**Reply by qingshuizheng** [reply]
Just bind "G" to a lambda function according to @DogLooksGood 's suggestion. Not sure whether to produce any caveats but currently working well:

`   '("G" . (lambda ()
             (interactive)
             (when (region-active-p)
              (meow-reverse))
             (meow-grab)))`
